cmd/olympus/actions/push.go

```diff
@@ -0,0 +1,59 @@
+package actions
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/gomods/athens/pkg/eventlog"
+	"github.com/gomods/athens/pkg/storage"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/worker"
+	"github.com/gomods/athens/pkg/payloads"
+)
+
+func pushNotificationHandler(w worker.Worker) func(c buffalo.Context) error {
+	return func(c buffalo.Context) error {
+		p := &payloads.PushNotification{}
+		if err := c.Bind(p); err != nil {
+			return err
+		}
+		pj, err := json.Marshal(p)
```

Comment:
I think this is fine for now, but for my information: why did you decide to bind and then marshal back to bytes again?

Reply:
i think worker is not working well with more complicated types, json is more safe.

Reply:
I'm not sure I follow @arschles . Do you mean why I don't just pass the request body to the worker?

Reply:
@marpio I was wondering why it calls `Bind` (which decodes bytes into `payloads.PushNotification{}`) and then immediately re-encodes the JSON back to bytes. I think I see now that the code is verifying that the JSON payload is correct. Is that right?